src/main/java/uk/gov/laa/ccms/caab/service/AssessmentService.java

```diff
@@ -0,0 +1,559 @@
+   */
+  protected boolean checkAssessmentForProceedingKeyChange(
+      final ApplicationDetail application,
+      final AssessmentEntityTypeDetail assessmentEntityType) {
+
+    if (assessmentEntityType != null && application.getProceedings() != null) {
+      for (final Proceeding proceeding : application.getProceedings()) {
+        final String matterType = proceeding.getMatterType().getId();
+        final String proceedingType = proceeding.getProceedingType().getId();
+        final String clientInvolvementType = proceeding.getClientInvolvement().getId();
+        final String entityId = getOpaInstanceMappingId(proceeding);
+
+        //find entity in entity type where matched entity id
+        final AssessmentEntityDetail assessmentEntity =
+            getAssessmentEntity(assessmentEntityType, entityId);
+
+        if (assessmentEntity != null) {
+          final AssessmentAttributeDetail matterTypeAttribute =
+              getAssessmentAttribute(assessmentEntity,
+                  AssessmentAttribute.MATTER_TYPE.getAttribute());
+          final AssessmentAttributeDetail proceedingTypeAttribute =
+              getAssessmentAttribute(assessmentEntity,
+                  AssessmentAttribute.PROCEEDING_NAME.getAttribute());
+          final AssessmentAttributeDetail clientInvolvementTypeAttribute =
+              getAssessmentAttribute(assessmentEntity,
+                  AssessmentAttribute.CLIENT_INVOLVEMENT_TYPE.getAttribute());
+
+          if (!matterType.equals(matterTypeAttribute.getValue())
+              || !proceedingType.equals(proceedingTypeAttribute.getValue())
+              || !clientInvolvementType.equals(clientInvolvementTypeAttribute.getValue())) {
+            return true;
+          }
+
+          //Check scope limitations for both, although not specifically needed for means, both
+          //assessments and application data should be kept in sync.
+          final AssessmentAttributeDetail scopeLimitationAttribute =
+              getAssessmentAttribute(assessmentEntity,
+                  AssessmentAttribute.REQUESTED_SCOPE.getAttribute());
+
+          if (scopeLimitationAttribute != null) {
+            final String assessmentScopeLimitation = scopeLimitationAttribute.getValue();
+            final String applicationScopeLimitation =
+                getRequestedScopeForAssessmentInput(proceeding);
+            if (!assessmentScopeLimitation.equals(applicationScopeLimitation)) {
+              return true;
+            }
+          }
+
+        } else {
+          return true;
+        }
+      }
+    } else {
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Retrieves the status of an assessment, defaulting to "Not Started" if the assessment is null.
+   *
+   * @param assessment the assessment to check the status of
+   * @return the status of the assessment or "Not Started" if assessment is null
+   */
+  private String getStatus(final AssessmentDetail assessment) {
```

Comment:
Would be good to return an AssessmentStatus here. Then you can do proper enum comparison.

Reply:
Amended